packages/service-library/src/servicelib/long_running_tasks/_task.py

```diff
@@ -405,9 +402,13 @@ def start_task(
     # only one unique task can be running
     if unique and tasks_manager.is_task_running(task_name):
         managed_tasks_ids = list(tasks_manager.get_task_group(task_name).keys())
-        assert len(managed_tasks_ids) == 1  # nosec
-        managed_task = tasks_manager.get_task_group(task_name)[managed_tasks_ids[0]]
-        raise TaskAlreadyRunningError(task_name=task_name, managed_task=managed_task)
+        if len(managed_tasks_ids) != 1:
+            raise RuntimeError(f"Unexpected amount of entries in {managed_tasks_ids=}")
+        task_id = managed_tasks_ids[0]
+        logger.info(
```

Comment:
you are aware that you are changing the behaviour of both fastapi and aiohttp long running tasks here?
Did you check if this was used somewhere else? e.g. in storage for example?

Reply:
Correct. Got confused since the ones in director-v2 are used by the webserver.
Never the less the interface was reverted and where required the the error is matched and the task_id is returned